src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+
+
+Let us review the example below where the buffer is allocated for the kernel's first (argument index 0) by using this API
+
+.. code:: c
+      :number-lines: 39
+           
+           xrtMemoryGroup idx_0 = xrtKernelArgGroupId(kernel, 0); // bank index of 0th argument
+           xrtBufferHandle a = xrtBOAlloc(device, data_size*sizeof(int), XCL_BO_FLAGS_NONE, idx_0);
+
+
+.. code:: c++
+      :number-lines: 15
+                       
+           auto input_buffer = xrt::bo(device, buffer_size_in_bytes, XCL_BO_FLAGS_NONE, kernel.group_id(0));
+
+
+
+The API fails if the kernel bank index is ambiguous. For example, the kernel has multiple CU with different connectivity for that argument. In those cases, it is required to create a kernel object/handle with specific a CU (or group of CUs with identical connectivity). 
+
+
+   
+Reading and write CU mapped registers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To read and write from the AXI-Lite register space corresponding to a CU, the CU must be opened in exclusive mode (in shared mode, multiple processes can access the CU's address space, hence it is unsafe if they are trying to access/change registers at the same time leading to a potential race behavior). The required APIs for kernel register read and write are
+  
+    - ``xrtKernelReadRegister`` (C++: Method ``xrt::kernel::read_register``)
+    - ``xrtKernelWriteRegiste`` (C++: Method ``xrt::kernel::write_register``)
+
+.. code:: c
+      :number-lines: 35
+         
+           int read_data; 
+           int write_data = 7; 
+              
+           xrtKernelHandle kernel = xrtPLKernelOpenExclusive(device, xclbin_uuid, "foo:{foo_1}");
+              
+           xrtKernelReadRegister(kernel,READ_OFFSET,&read_data);
+           xrtKernelWriteRegister(kernel,WRITE_OFFSET,write_data); 
+              
+           xrtKernelClose(kernel);
+
+
+In the above code block
+
+              - The compute unit named "foo_1" (name syntax: "kernel_name:{cu_name}") is opened exclusively.
+              - The Register Read/Write operation is performed. 
+              - Closed the kernel
+              
+**C++**: The equivalent C++ API example
+
+.. code:: c
+      :number-lines: 35
+       
+           int read_data; 
+           int write_data = 7; 
+              
+           auto krnl = xrt::kernel(device, "foo:{foo_1}", xclbin_uuid, true); 
+
+           read_data = kernel.read_register(READ_OFFSET);
+           kernel.write_register(WRITE_OFFSET,write_data); 
+              
+
+     
+Executing the kernel
+~~~~~~~~~~~~~~~~~~~~
+
+Execution of the kernel is associated with a **Run** handle (or object). The kernel can be executed by the API ``xrtKernelRun`` (in C++ overloaded operator ``xrt::kernel::()``) that takes all the kernel arguments in order. The kernel execution API returns a run handle (or object) corresponding to the execution. 
```

Comment:
```suggestion
Execution of the kernel is associated with a **Run** handle (or object). The kernel can be executed by the API ``xrtKernelRun`` (in C++ overloaded operator ``xrt::kernel::operator()``) that takes all the kernel arguments in order. The kernel execution API returns a run handle (or object) corresponding to the execution. 
```
By the way, keep the line short. Difficult to review otherwise...